packages/compass-connections/src/stores/connections-store-redux.ts

```diff
@@ -1701,6 +1742,36 @@ const connectWithOptions = (
           return;
         }
 
+        let showedNonRetryableErrorToast = false;
+        if (connectionInfo.atlasMetadata) {
```

Comment:
Do we need to gate this for atlas connections? Not that it makes a big difference, but adding logic branching hen not required can maybe be avoided

Not doing this and maybe updating our ws-proxy to return those in some cases might allow you to add e2e tests for that

Reply:
Nice suggestion, sounds good, removed the branching logic.

Added 2 unit tests, I'll work on an e2e test now.